src/puppetlabs/puppetdb/anonymizer.clj

```diff
@@ -350,6 +366,37 @@
         (update-in-nil ["containment_path"]
                        anonymize-containment-path newcontext config))))
 
+(pls/defn-validated anonymize-metric
+  [metric :- metric-schema-str
+   context
+   config]
+  (if (= "time" (get metric "category"))
+    (update-in metric ["name"] anonymize-leaf :type context config)
```

Comment:
So the metric resource name under `time` is actually in lower case, while the :type stuff expects the first letter to be upper case. This is creating a difference between the events and metrics area, ie. the random strings don't match - not even if you make the first letter lower case later on. I think you need to upcase the first letter of the input before getting something back, then lower it again or something like that - haven't really thought it through, perhaps something a bit more elegant :-).